tests/cli/test_buildspec.py

```diff
@@ -146,23 +146,35 @@ def test_buildspec_find_invalid():
     cache = BuildspecCache(configuration=configuration)
 
     # testing buildtest buildspec find invalid. This will assert SystemExit exception raised by sys.exit
-    with pytest.raises(SystemExit):
+    try:
         cache.print_invalid_buildspecs(error=True)
+    except SystemExit:
+        assert True
 
-    with pytest.raises(SystemExit):
+    try:
         cache.print_invalid_buildspecs(error=False)
+    except SystemExit:
+        assert True
 
-    with pytest.raises(SystemExit):
+    try:
         cache.print_invalid_buildspecs(error=True, terse=True)
+    except SystemExit:
+        assert True
```

Comment:
i think this is right thinking, what i would like to have is some getter method that can retrieve invalid specs when running `cache.print_invalid_ buildspecs`. Since we already have `self.cache["invalids"]` to get the invalids so if the key is empty we know there are no entry therefore we return from method without raising exception. What we don't have is a getter method. One has do something like this assuming `bc` is an object of type `BuildspecCache`

```
  if bc.cache['invalids']: 
    print("There are no invalid buildspecs")
```

We can have some method like `get_invalid_buildspecs` so one can just do `bc.get_invalid_buildspecs` to get a list of buildspecs that are invalid as a list.

Reply:
I guess we already have that function here 
https://github.com/buildtesters/buildtest/blob/devel/buildtest/cli/buildspec.py#:~:text=def%20get_invalid_buildspecs(,%5D.keys()